server/service/service_appconfig.go

```diff
@@ -110,175 +108,29 @@ func (svc *Service) ModifyAppConfig(ctx context.Context, p fleet.AppConfigPayloa
 	if err != nil {
 		return nil, err
 	}
-	config := appConfigFromAppConfigPayload(p, *oldAppConfig)
```

Comment:
Yes, I did the unmarshal into an already unmarshalled struct trick you mentioned. It wasn't ultra straight forward because of the needing to use `interface{}` but it worked.